platform/commonUI/edit/src/actions/EditAndComposeAction.js

```diff
@@ -54,9 +54,13 @@ define(
                         .then(doPersist);
             }
 
+            if (this.domainObject.getCapability('type').getKey() !== 'folder') {
+                this.domainObject.getCapability('action').perform('edit');
+            }
```

Comment:
Linking happens after editing - `doLink()` is the last thing called. Unless I'm misunderstanding what you're saying?

> Is there a better check than "is it a folder" for this?

Yeah I'm not wild about this either. We literally only want this behavior for folders though, so using any other check risks new types getting this behavior on drag-drop. Thought about putting the logic into a policy to make it a little more flexible, but seems like overkill for a v. simple piece of logic.

> First thing that comes to mind, which would solve both, is to check for the existence of an edit action. If one exists, perform it, and then add the link. Otherwise, just add the link.

This might work, but with the previous caveat. I'll do some investigation.